### ISSUE TITLE
toString broken in recursive function

### ISSUE THREAD

**Issue by alawatthe** [babel/babel#1164]
Using babel 5.0.8, the following code
```js
var factorial = function (n) {
  if (n === 0) {
    return 1;
  }
  else {
    return factorial(n - 1) * n;
  }
};
```
[gets transpiled to] (http://babeljs.io/repl/#?experimental=false&evaluate=false&loose=false&spec=false&playground=true&code=var%20factorial%20%3D%20function%20(n)%20%7B%0A%20%20if%20(n%20%3D%3D%3D%200)%20%7B%0A%20%20%20%20return%201%3B%0A%20%20%7D%0A%20%20else%20%7B%0A%20%20%20%20return%20factorial(n%20-%201)%20*%20n%3B%0A%20%20%7D%0A%7D%3B%0A)

```js
"use strict";

var factorial = (function (_factorial) {
  function factorial(_x) {
    return _factorial.apply(this, arguments);
  }

  factorial.toString = function () {
    return factorial.toString();        // <------- 
  };

  return factorial;
})(function (n) {
  if (n === 0) {
    return 1;
  } else {
    return factorial(n - 1) * n;
  }
});
```

which leads to a range error ```RangeError: Maximum call stack size exceeded``` when calling ```factorial.toString()```.

The line I marked should be
```js
  return _factorial.toString();
```

I think this was working before and the bug was introduced in [this commit](https://github.com/babel/babel/commit/cfff7aa6fb257cf39a85e7b06a5c49e8501ea32b).

The line 7 in the files https://github.com/babel/babel/blob/master/src/babel/transformation/templates/property-method-assignment-wrapper.js and https://github.com/babel/babel/blob/master/src/babel/transformation/templates/property-method-assignment-wrapper-generator.js
should be
```js
    return FUNCTION_KEY.toString();
```
instead of
```js
    return FUNCTION_ID.toString();
```

Some related tests need fixing, too.

Shall I create a pull request?

**Comment by eventualbuddha** [comment]
Seems like you're in the right track. A pull request would be great!

**Comment by sebmck** [comment]
Fixed as of 5.0.9, thanks!
